activesupport/test/testing/assertionless_tests_test.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+require_relative "../abstract_unit"
+
+module AssertionlessTestsTest
+  module Tests
+    def test_without_assertions
+    end
+
+    private
+      def with_assertionless_tests_behavior(behavior)
+        ActiveSupport::TestCase.assertionless_tests_behavior = behavior
+        yield
+      ensure
+        ActiveSupport::TestCase.assertionless_tests_behavior = :ignore
```

Comment:
I suppose `.with` was added for exactly this case:

```suggestion
ActiveSupport::TestCase.with(assertionless_tests_behavior: behavior) do
  yield
end
```

But then perhaps we could call it directly, without `with_assertionless_tests_behavior` helper. Totally up to you to decide

Reply:
Unfortunately, `with` will not work in this case, because when we assign a symbol (like :ignore), the `assertionless_tests_behavior` accessor now returns a proc. If we use `with`, we will reassign the previous value at the end (which is a proc), which will not work.

Reply:
Ah, right, makes sense 👍